src/runner/test_case_system.py

```diff
@@ -0,0 +1,26 @@
+import sys
+import yaml
+from src.framework.main_control_loop import MainControlLoop
+from src.runner.no_sim_test import NoSIM
+
+from src.sfr.state_field_registry import StateFieldRegistry
+
+class TestCaseSystem:
+    def run(self):
+        # creates necessary objects
+        if len(sys.argv) > 2:
+            # Read YAML file
+            file_path = sys.argv[1]
+        else:
+            file_path = 'src/config.yaml'
+
+        with open(file_path, 'r') as stream:
+            self.config = yaml.safe_load(stream)
+    
+        self.sfr = StateFieldRegistry()
+        self.mcl = MainControlLoop(self.config, self.sfr)
+        self.mcl.initialize()
+        self.mcl.default()
+        test = NoSIM(self.config, self.sfr, self.mcl)
```

Comment:
Can we upgrade TestCaseSystem to have the following structure: maybe a function like:

def create_test_case(self, test_case_name):
         self.test = python_magic_get_class(test_case_name)(self.config, self.sfr, self.mcl)
      
this separation will let us in the future accept the test case name as an argument